### ISSUE TITLE
Single instance mode

### ISSUE THREAD

**Issue by carbolymer** [loh-tar/wpa-cute#12]
Is there a way to run only a single instance of wpa-cute? I'm starting wpa-cute through a status icon on waybar, because I'm not always using wifi. So I'm often ending up with multiple wpa-cute instances running at the same time.

**Comment by loh-tar** [comment]
wpaCute don't do any session restore, or multi/single-instance stuff.

I can't say anything about waybar, guess some application start widget. You may have read some of my other comment https://github.com/loh-tar/wpa-cute/issues/9#issuecomment-642217984. In the meantime I could fix that trouble for me on KDE. There is an option in the KDE session configuration to ignore some application from restore.

> ...because I'm not always using wifi. 

Just a note: Not starting wpaCute do nothing to disable wifi! You can disable wifi e.g. by wpaCute and the Disconnect button. Or by some rfkill command, typically available by a special keyboard key. The Disconnect button tells wpa_supplicant not to try to find some network, rfkill power off the wifi card. IIRC
So, I would prefer to have the icon always in my system tray, where the icon tells me that wifi is disabled.

**Comment by carbolymer** [comment]
>wpaCute don't do any session restore, or multi/single-instance stuff.

Right. I guess it's more of a feature request then. 😄 

> So, I would prefer to have the icon always in my system tray, where the icon tells me that wifi is disabled.

I have more diagnostics information like, signal strength, interface IP, up&down data rate, frequency (I have the same SSID on multiple frequencies) on hover on my waybar widget, so I prefer it than having wpa-cute always in tray.

Maybe you would consider adding more diagnostics in the mouse hover info?

Anyway, this is more of a question if you have any suggestion how to handle this, than a bug request. I guess it should be possible with some shell script wrapper. Feel free to close this issue if you don't have a single-instance thing on your roadmap.

**Comment by loh-tar** [comment]
> I guess it should be possible with some shell script wrapper. 

Yes, that would have been my next suggestion ;-) But first look for such an ignore option. I think its the root of your trouble.

> Feel free to close this issue if you don't have a single-instance thing on your roadmap.

At the moment I have no idea how exactly I could implement this. Just checking if wpaCute is already running might not be enough. Maybe there are good reasons to have more than one instance running, 2x wifi or 1x wifi 1x lan cable? This enterprise stuff probably also works via cable...

> The notification from wpa-cute seems quirky for me

Yeah, I don't want to contradict that :-) It is virtually unchanged from the orig wpa_gui. I can't remember exactly if I ever tried to change it. But it could be. And then left it because the result was not satisfactory...

Ah, it's edited. In you original post you mentioned up&down data. I would also like to have this info. But this is not so easily possible. wpa_supplicant reveals all the information shown there, but does not the data rates. At least I haven't noticed them yet, see `wpa_cli status` and `wpa_cli bss 0` There is this est_throughput, but I haven't had a closer look yet.
All these other info could be quickly added to the status page. Feel free to dig deeper to the wpa_suplicant/wpa_cli docu and let me know how to fetch more info.

**Comment by loh-tar** [comment]
The tool-tip is changed in master, try it pls if its OK or if you would suggest more changes.
There is no need to install btw, just run the new compiled version.

**Comment by loh-tar** [comment]
Whoops! That's a bit embarrassing now...

> wpaCute don't do any session restore...

...I have just seen that there is a "hidden" piece of code packed in `#ifndef QT_NO_SESSIONMANAGER` that saves and restores a some values. I don't know if it does anything useful. I couldn't find a suitable looking config file quickly. Anyway, I should look into it :-/

**Comment by loh-tar** [comment]
New version is out, look at the new stuff, e.g. -R, and close this with some comment if this works for you now, and how.

> The notification from wpa-cute seems quirky for me... I need to right click on the icon, select status,

Ha! I forgot. middle-click was the trick :-)  I add a note the new man page

**Comment by carbolymer** [comment]
Thanks for the update! Hover notification is much more informative now. I haven't figured out single instance wrapper yet. Will update when I fix something together.
